### ISSUE TITLE
[server] please split `package` target into platform specific targets

### ISSUE THREAD

**Issue by SmartHoneybee** [mattermost/mattermost-server#8314]
#### Summary
It is currently not possible to build and package Mattermost only for one specific platform, instead one has to build (and package) Mattermost for all supported plattforms.

#### Steps to reproduce
`make config-reset build-linux package` (on a amd64 linux) results in an error due to a missing `${GOPATH}/bin/darwin_amd64/platform`.

#### Expected behavior
It would be nice to express that one is interested in building and packaging only a specific platform. E.g. with: `make config-reset build-linux package-linux`

#### Observed behavior (that appears unintentional) 
To achieve this, currently one has to:
* build binaries for all platforms, or
* edit a Makefile

#### Possible fixes
1. Split the `package` target into platform specific targets, similar to the `build` target.
2. Create a "common" target that prepares the packaging.

```patch
diff --git a/build/release.mk b/build/release.mk
index 5eaee80..026df8d 100644
--- a/build/release.mk
+++ b/build/release.mk
@@ -20,7 +20,9 @@ build-client:
 
 	cd $(BUILD_WEBAPP_DIR) && $(MAKE) build
 
-package:
+package: package-common package-osx package-windows package-linux
+
+package-common:
 	@ echo Packaging mattermost
 
 	@# Remove any old files
@@ -61,6 +63,7 @@ endif
 
 	@# ----- PLATFORM SPECIFIC -----
 
+package-osx: package-common
 	@# Make osx package
 	@# Copy binary
 ifeq ($(BUILDER_GOOS_GOARCH),"darwin_amd64")
@@ -73,6 +76,7 @@ endif
 	@# Cleanup
 	rm -f $(DIST_PATH)/bin/platform
 
+package-windows: package-common
 	@# Make windows package
 	@# Copy binary
 ifeq ($(BUILDER_GOOS_GOARCH),"windows_amd64")
@@ -85,6 +89,7 @@ endif
 	@# Cleanup
 	rm -f $(DIST_PATH)/bin/platform.exe
 
+package-linux: package-common
 	@# Make linux package
 	@# Copy binary
 ifeq ($(BUILDER_GOOS_GOARCH),"linux_amd64")
```

**Comment by SmartHoneybee** [comment]
To demonstrate that implementing this does not change the current behavior, a simplified example:
```shell
$ cat Makefile 
package: common windows darwin

common:
	@echo common

windows: common
	@echo windows

darwin: common
	@echo darwin
$ make package
common
windows
darwin
$ make windows
common
windows
```

**Comment by ccbrown** [comment]
I don't think you'll find any opposition if you were to put in a PR. :)

**Comment by lindy65** [comment]
Hey @SmartHoneybee,

I'll close this issue off here for now ...

**Comment by SmartHoneybee** [comment]
Hi @lindy65,

thank you!

--
Note to future me: fixed by #8366
